Sources_v3/Controllers/CurrentUserController/CurrentUserController.swift

```diff
@@ -316,6 +316,85 @@ public extension _CurrentChatUserController {
             }
         }
     }
+    
+    /// Registers a device to the current user.
+    /// `setUser` must be called before calling this.
+    /// - Parameters:
+    ///   - token: Device token, obtained via `didRegisterForRemoteNotificationsWithDeviceToken` function in `AppDelegate`.
+    ///   - completion: Called when device is successfully registered, or with error.
+    func addDevice(token: Data, completion: ((Error?) -> Void)? = nil) {
+        guard let currentUserId = currentUser?.id else {
+            completion?(ClientError.CurrentUserDoesNotExist())
```

Comment:
Shouldn't we always call completions on `callbackQueue`?

Reply:
Fixed in 7d033b231fcd6938db52b3e9c4bb77f55b524e5b